server/src/models/pmcmean.rs

```diff
@@ -61,7 +146,234 @@ pub fn grid(
     }
 }
 
-/** Private Functions **/
-fn decode(model: &[u8]) -> f32 {
+/// Read the coefficient for a model of type PMC-Mean from a slice of bytes. The
+/// coefficient is the average value of the time series segment whose vales are
+/// represented by the model.
+fn decode_model(model: &[u8]) -> f32 {
     f32::from_be_bytes(model.try_into().unwrap())
 }
+
+/// Returns true if `v1` and `v2` are equivalent or both values are NAN.
+fn equal_or_nan(v1: f32, v2: f32) -> bool {
+    v1 == v2 || (v1.is_nan() && v2.is_nan())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use proptest::num::f32;
+    use proptest::{prop_assert, prop_assume, proptest};
+
+    // Tests for PMCMeanModelType.
+    proptest! {
```

Comment:
Maybe indent tests within a "proptest" scope. If this is changed here it also needs to be changed under.

Reply:
Unfortunately `rustfmt` currently cannot format code inside macros #rust-lang/rustfmt/issues/8 and `proptest` does not provide other means for defining property-based tests #AltSysrq/proptest/issues/149. So, although I agree that the current formatting is less than optimal, I propose we keep it as is to not spend time fighting `rustfmt`.